cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -217,27 +218,49 @@ fn insert_load_store(
 ) -> Result<()> {
     let ctrl_type = *rets.first().or(args.first()).unwrap();
     let type_size = ctrl_type.bytes();
-    let (address, offset) = fgen.generate_load_store_address(builder, type_size)?;
+
+    // AArch64 has issues with unaligned atomics.
+    // https://github.com/bytecodealliance/wasmtime/issues/5483
+    let is_atomic = [Opcode::AtomicLoad, Opcode::AtomicStore].contains(&opcode);
+    let max_offset = if is_atomic && cfg!(target_arch = "aarch64") {
+        0
+    } else {
+        128
+    };
+    let (address, offset) = fgen.generate_load_store_address(builder, type_size, max_offset)?;
 
     // TODO: More advanced MemFlags
     let flags = MemFlags::new();
 
     // The variable being loaded or stored into
     let var = fgen.get_variable_of_type(ctrl_type)?;
 
-    if opcode.can_store() {
-        let val = builder.use_var(var);
+    match opcode {
```

Comment:
Let's replace this with `match opcode.format()`. Then the cases below are, in order: `InstructionFormat::LoadNoOffset`, `InstructionFormat::StoreNoOffset`, `InstructionFormat::Store`, `InstructionFormat::Load`. Any other format should panic with `unimplemented!()`. That captures what we actually wanted to know about the opcode: we don't care whether it's atomic or whether it's a store, we just need to know how to construct the right `InstructionData`.

This also highlights that the `offset` value returned from `generate_load_store_address` is unused for these atomic ops. I expect to get slightly better fuzzing results if we avoid generating a random number that we then ignore. How about returning a `max_offset` instead and only doing the `int_in_range` call in the `Store` and `Load` format cases?

Reply:
Those are great suggestions, Thanks!